ServiceFabric.PubSubActors/Helpers/BrokerServiceLocator.cs

```diff
@@ -41,25 +42,61 @@ public BrokerServiceLocator(bool useRemotingV2 = false)
         public async Task RegisterAsync(Uri brokerServiceName)
         {
             var activationContext = FabricRuntime.GetActivationContext();
-            var fc = new FabricClient();
-            await fc.PropertyManager.PutPropertyAsync(new Uri(activationContext.ApplicationName), nameof(BrokerService), brokerServiceName.ToString());
+            await _fabricClient.PropertyManager.PutPropertyAsync(new Uri(activationContext.ApplicationName), _borkerName, brokerServiceName.ToString());
         }
 
         /// <inheritdoc />
         public async Task<Uri> LocateAsync()
         {
             try
             {
+                // check current context
                 var activationContext = FabricRuntime.GetActivationContext();
-                var fc = new FabricClient();
-                var property = await fc.PropertyManager.GetPropertyAsync(new Uri(activationContext.ApplicationName), nameof(BrokerService));
-                if (property == null) return null;
-                string value = property.GetValue<string>();
-                return new Uri(value);
+                var property = await GetBrokerPropertyOrNull(activationContext.ApplicationName);
+                
+                if (property == null)
+                {
+                    // try to find broker name in other application types
+                    var apps = await _fabricClient.QueryManager.GetApplicationListAsync();
```

Comment:
this should use the paging feature

Reply:
didn't know for paging feature, thanks, implemented in last change